### ISSUE TITLE
Can you help me to write a ddraw.dll for my Game?

### ISSUE THREAD

**Issue by 515to** [CnCNet/ts-ddraw#10]
Can you help me to write a ddraw.dll for my Game?
This is my game client download url:
https://mega.nz/#!aWxWCaSJ!6joDY6pjBGkSJV5irv2qjzEXKL4--uQo37XWyusGuxU
 when game runs on Win8 or higher , its fps is below 30 or less. and AMD video card, game is black screen. 
when it runs on Win7 , its fps always  around 200 or more.

i can pay for it , and tell me how much. thank you.
my skype: nashine@live.cn

**Comment by dkeetonx** [comment]
Have you tried cnc-ddraw? https://github.com/CnCNet/cnc-ddraw

What's the name of the game?

**Comment by 515to** [comment]
I have tried, the name of the game is TalesWeaver.
when i started game with ddraw.dll from CnCNet, it shows "Unable to find an entry point named 'DirectDrawCreateEx' in DLL 'ddraw.dll'"
Thank you

**Comment by elishacloud** [comment]
@515to, have you tried this project: [DDrawCompat](https://github.com/narzoul/DDrawCompat).  It has fixes for these types of issues.  Also I have created a project to help with these kind of issues also.  See: [dxwrapper](https://github.com/elishacloud/dxwrapper)

**Comment by FunkyFr3sh** [comment]
Neither ts-ddraw nor cnc-ddraw support DirectX 7. I don't think there is any point in adding support for it since I'm pretty sure all games using DirectX 7 will also take advantage of Direct3D which can't be supported by our projects (We got standalone solutions that don't rely on the old ddraw/d3d dlls shipped with windows).

**Comment by elishacloud** [comment]
I have tested close to 100 games that use ddraw.  Most DirectX7 games use Direct3D.  However I found 6 games that use DirectX 7 without taking advantage of Direct3D, or at least without taking advantage of any Direct3DDevice APIs.  Some of them use APIs like IDirect3D::EnumDevices or IDirect3D::FindDevice, which are pretty easy to emulate with Direct3D9 without building all the Direct3D support.

Here is a list of games that use only ddraw in DirectX7:
* Age of Wonders 2
* Atlantis 3 The New World
* Gangsters Organized Crime
* Heroes of Might and Magic 4
* Majesty Gold HD (Old version)
* Stronghold Crusader Extreme HD

With my [dxwrapper](https://github.com/elishacloud/dxwrapper) project I am trying to do full conversion of ddraw (including Direct3D) into Direct3D9.
